File: src/main/java/Piruru/cards/DiscardAndGainEnergy.java
package Piruru.cards;

import Piruru.abstracts.PiruruCard;
import com.megacrit.cardcrawl.actions.common.DiscardAction;
import com.megacrit.cardcrawl.actions.common.GainEnergyAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

import static Piruru.Piruluk.makeID;

public class DiscardAndGainEnergy extends PiruruCard {
    private static final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(
            makeID(DiscardAndGainEnergy.class));
    private static final int COST = 0;
    private static final CardType TYPE = CardType.SKILL;
    private static final CardRarity RARITY = CardRarity.COMMON;
    private static final CardTarget TARGET = CardTarget.SELF;
    private static final int ENERGY = 1;
    private static final int NRG_UP = 1;

    public DiscardAndGainEnergy() {
        super(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, NRG_UP, COST);
        baseMagicNumber = magicNumber = ENERGY;
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        act(new DiscardAction(p, p, 1, false));
        act(new GainEnergyAction(magicNumber));
    }
}

File: src/main/kotlin/Piruru/cards/DiscardAndGainEnergy.kt
package Piruru.cards

import Piruru.Piruluk.Statics.makeID
import Piruru.abstracts.PiruruCard
import com.megacrit.cardcrawl.actions.common.DiscardAction
import com.megacrit.cardcrawl.actions.common.GainEnergyAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.localization.CardStrings
import com.megacrit.cardcrawl.monsters.AbstractMonster


class DiscardAndGainEnergy : PiruruCard(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, NRG_UP, COST) {
    init {
        magicNumber = ENERGY
        baseMagicNumber = magicNumber
    }

    override fun use(p: AbstractPlayer, m: AbstractMonster?) {
        act(DiscardAction(p, p, 1, false))
        act(GainEnergyAction(magicNumber))
    }

    companion object {
        private val cardStrings = CardCrawlGame.languagePack.getCardStrings(
                makeID(DiscardAndGainEnergy::class.java))
        private val COST = 0
        private val TYPE = AbstractCard.CardType.SKILL
        private val RARITY = AbstractCard.CardRarity.COMMON
        private val TARGET = AbstractCard.CardTarget.SELF
        private val ENERGY = 1
        private val NRG_UP = 1
    }
}
